lookout/style/format/benchmarks/generate_styles.py

```diff
@@ -0,0 +1,114 @@
+"""A module for repositories generation with different styles in different branches."""
+import logging
+from pathlib import Path
+import re
+import shutil
+import subprocess
+import sys
+import tarfile
+from typing import Tuple
+
+from lookout.core import slogging
+
+
+class _NoChangesException(Exception):
+    """Exception should be raised if no changes were made to initial code."""
+
+    pass
+
+
+js_operators = [
+    "+", "-", "*", "/", "%", "++", "--",
+    "=", "+=", "-=", "/=", "%=",
+    "==", "===", "!=", "!==", ">", "<", ">=", "<=",
+    "&", "|", "~", "^", ">>", "<<", ">>>",
+    "&&", "||", "!",
+    "?", ":",
+]
+
+js_operators_regex = "|".join(re.escape(op) for op in sorted(js_operators, reverse=True))
+
+js_styles_regex = {
+    "equal_no_space_style": (" = ", "="),
+    "operators_no_space_style": (" (%s) " % js_operators_regex, r"\1"),
+    "no_newline_before_function": ("(\n+)function ", "\nfunction "),
+    "two_newlines_before_function": ("(\n+)function ", "\n\nfunction "),
+    "no_space_after_if": (r"if\s*\(", r"if("),
+    "no_spaces_inside_brackets": (r"\{ +([^{}]+) +\}", r"{\1}"),
+    "no_spaces_inside_round_brackets": (r"\( +([^()]+) +\)", r"(\1)"),
+    "spaces_inside_round_brackets": (r"\(([^()]+)\)", r"( \1 )"),
+}
```

Comment:
This may be enough for very basic changes (even though those listed here will already be super weird in the sense that they don't distinguish between function `{}`-s and objects `{}`-s, same for parens and brackets, etc) but if we want to do anything even remotely more complicated we should use ASTs to transform formatting, not regexes.

Reply:
Yes, but it is to make it faster now, I also do not like it. I think to have two options here: 
1. regexp for simple cases. 
2. Any patch function you want otherwise. 
If you want I can make a TODO to remember about it :)